src/pages/home/report/ReportActionCompose.js

```diff
@@ -323,14 +358,86 @@ class ReportActionCompose extends React.Component {
-     *
-     * @param {Boolean} hasExceededMaxCommentLength
+     * Clean data related to EmojiSuggestions
      */
-    setExceededMaxCommentLength(hasExceededMaxCommentLength) {
-        this.setState({hasExceededMaxCommentLength});
+    resetSuggestedEmojis() {
+        this.setState({
+            suggestedEmojis: [],
+            shouldShowSuggestionMenu: false,
+        });
+    }
+
+    /**
+     * Calculates and cares about the content of an Emoji Suggester
+     */
+    calculateEmojiSuggestion() {
+        const leftString = this.state.value.substring(0, this.state.selection.end);
+        const colonIndex = leftString.lastIndexOf(':');
+        const isCurrentlyShowingEmojiSuggestion = this.isEmojiCode(this.state.value, this.state.selection.end);
+
+        // the larger composerHeight the less space for EmojiPicker, Pixel 2 has pretty small screen and this value equal 5.3
+        const hasEnoughSpaceForLargeSuggestion = this.props.windowHeight / this.state.composerHeight >= 6.8;
+        const isEmojiPickerLarge = !this.props.isSmallScreenWidth || (this.props.isSmallScreenWidth && hasEnoughSpaceForLargeSuggestion);
+
+        const nextState = {
+            suggestedEmojis: [],
+            highlightedEmojiIndex: 0,
+            colonIndex,
+            shouldShowSuggestionMenu: false,
+            isEmojiPickerLarge,
+        };
+        const newSuggestedEmojis = EmojiUtils.suggestEmojis(leftString);
+
+        if (newSuggestedEmojis.length && isCurrentlyShowingEmojiSuggestion) {
+            nextState.suggestedEmojis = newSuggestedEmojis;
+            nextState.shouldShowSuggestionMenu = !_.isEmpty(newSuggestedEmojis);
+        }
+
+        LayoutAnimation.configureNext(LayoutAnimation.create(50, LayoutAnimation.Types.easeInEaseOut, LayoutAnimation.Properties.opacity));
+
+        this.setState(nextState);
+    }
+
+    /**
+     * Check if this piece of string looks like an emoji
+     * @param {String} str
+     * @param {Number} pos
+     * @returns {Boolean}
+     */
+    isEmojiCode(str, pos) {
+        const leftWords = str.slice(0, pos).split(CONST.REGEX.NEW_LINE_OR_WHITE_SPACE);
+        const leftWord = _.last(leftWords);
+
+        return CONST.REGEX.HAS_COLON_ONLY_AT_THE_BEGINNING.test(leftWord) && leftWord.length > 2;
+    }
+
+    /**
+     * Replace the code of emoji and update selection
+     * @param {Number} highlightedEmojiIndex
+     */
+    insertSelectedEmoji(highlightedEmojiIndex) {
+        const commentBeforeColon = this.state.value.slice(0, this.state.colonIndex);
+        const emojiObject = this.state.suggestedEmojis[highlightedEmojiIndex];
+        const emojiCode = emojiObject.types && emojiObject.types[this.props.preferredSkinTone] ? emojiObject.types[this.props.preferredSkinTone] : emojiObject.code;
+        const commentAfterColonWithEmojiNameRemoved = this.state.value.slice(this.state.selection.end).replace(CONST.REGEX.EMOJI_REPLACER, CONST.SPACE);
```

Comment:
@perunt Is there a specific reason why we are replacing text after colon with a space? (Coming from https://github.com/Expensify/App/issues/19289)